petab/core.py

```diff
@@ -900,3 +900,87 @@ def measurements_have_replicates(measurement_df: pd.DataFrame) -> bool:
             measurement_df,
             ['observableId', 'simulationConditionId',
              'preequilibrationConditionId', 'time'])).size().values - 1)
+
+
+def flatten_timepoint_specific_output_overrides(
+        petab_problem: Problem) -> None:
+    """If the PEtab problem definition has timepoint-specific
+    observableParameters or noiseParameters for the same observable, replace
+    those by replicating the respective observable.
+
+    This is a helper function for some tools which may not support such
+    timepoint-specific mappings.
+
+    Arguments:
+        petab_problem:
+            PEtab problem to work on
+    """
+
+    df = petab_problem.measurement_df[
+        ["observableId",
+         "preequilibrationConditionId",
+         "simulationConditionId"]
+    ]
+    df_unique_values = df.drop_duplicates()
+    del df  # remove variable
+
+    # Create empty df -> to be filled with replicate-specific observables
+    df_new = pd.DataFrame()
+
+    for nrow in range(len(df_unique_values.index)):
+        df = petab_problem.measurement_df.loc[
```

Comment:
add comment: what does this do? are all rows selected that have a specific set of {observableId,preeqCondId,simCondId}?